File: core/src/main/java/ar/com/phostech/vertx/core/env/RuntimeEnvironment.java
package ar.com.phostech.vertx.core.env;

/**
 * @author luis
 * Date 23/02/18 22:14
 * Project: vertxms
 */
public interface RuntimeEnvironment {

    /**
     * This implementation returns a default implementation for a
     * RuntimeEnvironment
     *
     * @return RuntimeEnvironment instance
     */
    static RuntimeEnvironment get() {
        return ExecutionEnvironment.INSTANCE;
    }

    /**
     * Returns true if the environment is local (dev)
     *
     * @return true if the running environment is dev
     */
    boolean development();

    /**
     * Returns the application name
     *
     * @return
     */
    String application();

    /**
     * Returns application scope
     *
     * @return
     */
    String scope();

}

File: core/src/main/kotlin/ar/com/phostech/vertx/core/env/RuntimeEnvironment.kt
package ar.com.phostech.vertx.core.env

interface RuntimeEnvironment {

    /**
     * Returns true if the environment is development
     *
     * @return true if the running environment is dev
     */
    fun development(): Boolean

    /**
     * Returns the application name
     *
     * @return
     */
    fun application(): String

    /**
     * Returns application scope
     *
     * @return
     */
    fun scope(): String
}
